### PR TITLE
GRIDEDIT-977 Correct DisposableGriddedSamples and GriddedSamplesNative

### PR DESCRIPTION
**Pull request by lucacarniato**
Introduce API for getting interpolation types

### CODE REVIEW

File: src/MeshKernelNET/Native/GriddedSamplesNative.cs [hunk 1]
```diff
@@ -48,13 +48,8 @@ public struct GriddedSamplesNative
         public IntPtr values;
 
         /// <summary>
-        /// X coordinate of the upper right
+        /// The numeric representation of the values (0 = short, 1 = float, 2 = int and 3 = double)
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Native/GriddedSamplesNative.cs:51]: Can this be made an enum?

File: src/MeshKernelNET/Api/DisposableGriddedSamples.cs [hunk 2]
```diff
@@ -126,8 +116,7 @@ protected override void SetNativeObject(ref GriddedSamplesNative nativeObject)
             nativeObject.coordinates_x = GetPinnedObjectPointer(coordinatesX);
             nativeObject.coordinates_y = GetPinnedObjectPointer(coordinatesY);
             nativeObject.values = GetPinnedObjectPointer(values);
-            nativeObject.upper_right_x = upperRightX;
-            nativeObject.upper_right_y = upperRightY;
+            nativeObject.values = GetPinnedObjectPointer(values);
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Api/DisposableGriddedSamples.cs]: copy-paste error? duplicate line, and valueType is not set! If this wasn't detected by unit tests also a unit test is missing.

File: src/MeshKernelNET/Native/MeshKernelDll.cs [hunk 3]
```diff
@@ -832,6 +832,38 @@ static MeshKernelDll()
         [DllImport(MeshKernelDllName, EntryPoint = "mkernel_get_projection_spherical_accurate", CallingConvention = CallingConvention.Cdecl)]
         internal static extern int GetProjectionSphericalAccurate([In][Out] ref int projection);
 
+        /// <summary>
+        /// Gets an int indicating the interpolation type short
+        /// </summary>
+        /// <param name="projection">The int indicating the short type</param>
+        /// <returns>Error code</returns>
+        [DllImport(MeshKernelDllName, EntryPoint = "mkernel_get_interpolation_type_short", CallingConvention = CallingConvention.Cdecl)]
+        internal static extern int GetInterpolationTypeShort([In][Out] ref int type);
```
Comment (andreasbuykx) [inline src/MeshKernelNET/Native/MeshKernelDll.cs:841]: An enum is more user-friendly

File: src/MeshKernelNET/Api/DisposableGriddedSamples.cs (hunk 2 above)
Comment (lucacarniato) [reply src/MeshKernelNET/Api/DisposableGriddedSamples.cs]: this is the native struct   

 struct GriddedSamples
    {
        /// @brief Number of x gridded samples coordinates
        int num_x = 0;

        /// @brief Number of y gridded samples coordinates
        int num_y = 0;

        /// @brief X coordinate of the grid origin (lower left corner)
        double x_origin = 0.0;

        /// @brief Y coordinate of the grid origin (lower left corner)
        double y_origin = 0.0;

        /// @brief Constant grid cell size
        double cell_size = 0.0;

        /// @brief If not nullptr, coordinates for non-uniform grid spacing in x direction
        double* x_coordinates = nullptr;

        /// @brief If not nullptr, coordinates for non-uniform grid spacing in y direction
        double* y_coordinates = nullptr;

        /// @brief Sample values
        void* values = nullptr;

        /// @brief the numeric representation of the values (0 = short, 1 = float)
        int value_type = 1;
    };

**Review by andreasbuykx — APPROVED** [review]

**Pull request merged**